packages/time-utilities/src/lib/Cron.ts

```diff
@@ -0,0 +1,138 @@
+				return new Date(Date.UTC(outset.getUTCFullYear(), outset.getUTCMonth(), outset.getUTCDate(), hour, minute));
+			}
+		}
+
+		return this.next(new Date(outset.getTime() + Time.Day), false);
+	}
+
+	/**
+	 * Normalize the pattern
+	 * @param cron The pattern to normalize
+	 */
+	private static normalize(cron: string): string {
+		if (cron in predefined) return predefined[cron];
+		const now = new Date();
+		cron = cron
+			.split(' ')
+			.map((val, i) =>
+				val.replace(wildcardRegex, (match) => {
+					if (match === 'h') return (Math.floor(Math.random() * allowedNum[i][1]) + allowedNum[i][0]).toString();
+
+					if (match === '?') {
+						switch (i) {
+							case 0:
+								return now.getUTCMinutes().toString();
+							case 1:
+								return now.getUTCHours().toString();
+							case 2:
+								return now.getUTCDate().toString();
+							case 3:
+								return now.getUTCMonth().toString();
+							case 4:
+								return now.getUTCDay().toString();
+						}
+					}
+
+					return match;
+				})
+			)
+			.join(' ');
+		return cron.replace(tokensRegex, (match) => String(cronTokens[match]));
+	}
+
+	/**
+	 * Parse the pattern
+	 * @param cron The pattern to parse
+	 */
+	private static parseString(cron: string): Array<number[]> {
+		const parts = cron.split(' ');
+		if (parts.length !== 5) throw new Error('Invalid Cron Provided');
+		return parts.map((part, i) => Cron.parsePart(part, i));
+	}
+
+	/**
+	 * Parse the current part
+	 * @param cronPart The part of the pattern to parse
+	 * @param id The id that identifies the current part
+	 */
+	private static parsePart(cronPart: string, id: number): number[] {
+		if (cronPart.includes(',')) {
+			const res = [];
+			for (const part of cronPart.split(',')) res.push(...Cron.parsePart(part, id));
+			return [...new Set(res)].sort((a, b) => a - b);
+		}
+
+		// eslint-disable-next-line prefer-const
+		const [, wild, minStr, maxStr, step] = partRegex.exec(cronPart) as RegExpExecArray;
+		let [min, max] = [parseInt(minStr, 10), parseInt(maxStr, 10)];
+
+		// If '*', set min and max as the minimum and maximum allowed numbers:
+		if (wild) [min, max] = allowedNum[id];
+		// Else if a number was given, but not a maximum nor a step, return it
+		// as only allowed value:
+		else if (!max && !step) return [min];
+
+		// Set min and max as the given numbers, defaulting max to the maximum
+		// allowed, so min is never bigger than max:
+		// This makes min and max be, in the following cases (considering minutes):
+		// -> 1-2 | 1..2
+		// -> 2-1 | 1..2
+		// -> 1/7 | 1, 8, 15, 22, 29, 36, 43, 50, 57
+		[min, max] = [min, max || allowedNum[id][1]].sort((a, b) => a - b);
+
+		// Generate a range
+		return Cron.range(min, max, parseInt(step, 10) || 1);
+	}
+
+	/**
+	 * Get an array of numbers with the selected range
+	 * @param min The minimum value
+	 * @param max The maximum value
+	 * @param step The step value
+	 */
+	private static range(min: number, max: number, step: number): number[] {
```

Comment:
This should be moved to `@sapphire/utilities` imo.. Thoughts @sapphire-project/developers?

Reply:
What is your reasoning Vladdy? Cron feels time related to me.

Reply:
I'm pretty sure Vladdy's referring to the range method itself being moved to `@sapphire/utilities` to be used as a general utility function.

Reply:
^

Reply:
Oh, yes that would be good